sympy/physics/control/lti.py

```diff
@@ -580,20 +593,29 @@ class Series(Basic):
 
     """
     def __new__(cls, *args, evaluate=False):
-        if not all(isinstance(arg, (TransferFunction, Parallel, Series)) for arg in args):
+        if not all(isinstance(arg, (TransferFunction, TransferFunctionMatrix, Parallel, Series))
+            for arg in args):
             raise TypeError("Unsupported type of argument(s) for Series.")
 
         obj = super(Series, cls).__new__(cls, *args)
-        obj._var = None
-        for arg in args:
-            if obj._var is None:
-                obj._var = arg.var
-            elif obj._var != arg.var:
-                raise ValueError("All transfer functions should use the same complex"
-                    " variable of the Laplace transform.")
-        if evaluate:
-            return obj.doit()
-        return obj
+        obj.is_SISO = all(isinstance(arg.doit(), TransferFunction) for arg in args)
+        if not obj.is_SISO:
+            obj._num_outputs, obj._num_inputs = args[0].num_outputs, args[-1].num_inputs
+            for x in range(len(args) - 1):
+                # input-output sizes should be consistent.
+                if args[x].num_inputs != args[x + 1].num_outputs:
+                    raise ValueError("Argument {0} of Series has {1} input(s),"
+                        " but argument {2} has {3} output(s)."
+                        .format(x + 1, args[x].num_inputs, x + 2, args[x + 1].num_outputs))
+        else:
+            obj._num_outputs, obj._num_inputs = 1, 1
+
+        tf = "transfer functions" if obj.is_SISO else "TransferFunctionMatrix objects"
+        obj._var = args[0].var
+        if not all(arg.var == obj._var for arg in args):
+            raise ValueError("All {0} should use the same complex"
+                " variable of the Laplace transform.".format(tf))
+        return obj.doit() if evaluate else obj
```

Comment:
This `[0]` causes `Series()` with no arguments to crash

Reply:
@namannimmo10 You should also add test cases for these after resolving this.